source/code/scxsystemlib/disk/staticlogicaldiskenumeration.cpp

```diff
@@ -109,6 +109,65 @@ namespace SCXSystemLib
         }
     }
 
+    /*----------------------------------------------------------------------------*/
+    /**
+       Update the enumeration for specific instance.
+
+       \param updateInstances If true (default) all instances will be updated.
+                              Otherwise only the content of teh enumeration will be updated.
+
+    */
+
+    void StaticLogicalDiskEnumeration::UpdateSpecific(bool updateInstances/*=true*/, std::wstring mountPoint, size_t *pos)
+    {
+        SCX_LOGTRACE(m_log, SCXCoreLib::StrAppend(L"Size of enumeration: ", this->Size()));
+        for (EntityIterator iter=Begin(); iter!=End(); ++iter)
+        {
+            SCXCoreLib::SCXHandle<StaticLogicalDiskInstance> disk = *iter;
+            if ( mountPoint != L"" && disk->m_mountPoint != mountPoint) continue;
+            SCX_LOGTRACE(m_log, SCXCoreLib::StrAppend(L"Device being set to OFFLINE, disk: ", disk->m_mountPoint));
+            disk->m_online = false;
```

Comment:
if mountPoint == L"" then also disk->m_online is set to false. Confirm if it is correct behavior?

Reply:
It is correct behavior. It will be set to online afterwards